queue_services/entity-digital-credentials/src/entity_digital_credentials/config.py

```diff
@@ -0,0 +1,166 @@
+# Copyright © 2023 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""All of the configuration for the service is captured here.
+
+All items are loaded, or have Constants defined here that
+are loaded into the Flask configuration.
+All modules and lookups get their configuration from the
+Flask config, rather than reading environment variables directly
+or by accessing this configuration directly.
+"""
+import os
+import random
+
+from dotenv import find_dotenv, load_dotenv
+
+
+# this will load all the envars from a .env file located in the project root (api)
+load_dotenv(find_dotenv())
+
+CONFIGURATION = {
+    'development': 'legal_api.config.DevConfig',
+    'testing': 'legal_api.config.TestConfig',
+    'production': 'legal_api.config.ProdConfig',
+    'default': 'legal_api.config.ProdConfig'
+}
+
+
+def get_named_config(config_name: str = 'production'):
+    """Return the configuration object based on the name.
+
+    :raise: KeyError: if an unknown configuration is requested
+    """
+    if config_name in ['production', 'staging', 'default']:
+        config = ProdConfig()
+    elif config_name == 'testing':
+        config = TestConfig()
+    elif config_name == 'development':
+        config = DevConfig()
+    else:
+        raise KeyError(f'Unknown configuration: {config_name}')
+    return config
+
+
+class _Config():  # pylint: disable=too-few-public-methods
+    """Base class configuration that should set reasonable defaults.
+
+    Used as the base for all the other configurations.
+    """
+
+    PROJECT_ROOT = os.path.abspath(os.path.dirname(__file__))
+
+    MSG_RETRY_NUM = int(os.getenv('MSG_RETRY_NUM', '5'))
+
+    LD_SDK_KEY = os.getenv('LD_SDK_KEY', None)
+
+    # urls
+    LEGAL_API_URL = os.getenv('LEGAL_API_URL', None)
+
+    # variables
+    LEGISLATIVE_TIMEZONE = os.getenv(
+        'LEGISLATIVE_TIMEZONE', 'America/Vancouver')
+
+    SQLALCHEMY_TRACK_MODIFICATIONS = False
+
+    # POSTGRESQL
+    DB_USER = os.getenv('DATABASE_USERNAME', '')
+    DB_PASSWORD = os.getenv('DATABASE_PASSWORD', '')
+    DB_NAME = os.getenv('DATABASE_NAME', '')
+    DB_HOST = os.getenv('DATABASE_HOST', '')
+    DB_PORT = os.getenv('DATABASE_PORT', '5432')
```

Comment:
we will be used a shared config in 1pwd for this so the env var names we will be pulling from will match what entity-bn is using.

https://github.com/bcgov/lear/blob/f61c84b50887c3c5e894169743ee0cf3bfdfdde7/queue_services/entity-bn/src/entity_bn/config.py#L84-L88